frontend/server/src/Controllers/User.php

```diff
@@ -2512,6 +2514,65 @@ function () use (
         );
     }
 
+    /**
+     * Get authors rank.
+     *
+     * @return AuthorsRank
+     */
+    public static function getAuthorsRank(
+        int $offset,
+        int $rowCount
+    ): array {
+        return \OmegaUp\Cache::getFromCacheOrSet(
+            \OmegaUp\Cache::AUTHORS_RANK,
+            "{$offset}-{$rowCount}",
+            /**
+             * @return AuthorsRank
+             */
+            function () use (
+                $offset,
+                $rowCount
+            ): array {
+                return \OmegaUp\DAO\UserRank::getAuthorsRank(
+                    $offset,
+                    $rowCount
+                );
+            },
+            APC_USER_CACHE_USER_RANK_TIMEOUT
+        );
+    }
+
+    /**
+     * Prepare all the properties to be sent to the
+     * author rank table view via smarty.
+     *
+     * @return array{smartyProperties: array{payload: AuthorRankTablePayload}, entrypoint: string}
+     *
+     * @omegaup-request-param int $length
+     * @omegaup-request-param int $page
+     */
+    public static function getAuthorRankForSmarty(\OmegaUp\Request $r) {
+        $r->ensureInt('page', null, null, false);
+        $r->ensureInt('length', null, null, false);
+
+        $page = is_null($r['page']) ? 1 : intval($r['page']);
+        $length = is_null($r['length']) ? 100 : intval($r['length']);
+
+        return [
+            'smartyProperties' => [
+                'payload' => [
+                    'page' => $page,
+                    'length' => $length,
+                    'ranking' => self::getAuthorsRank(
+                        $page,
+                        $length
```

Comment:
Después de https://github.com/omegaup/omegaup/pull/3898, debería ser posible escribir esto como:

```suggestion
                        $r->ensureOptionalInt('page') ?? 1; 
                        $r->ensureOptionalInt('length') ?? 100;
```